src/toil/provisioners/clusterScaler.py

```diff
@@ -28,6 +28,17 @@
 
 logger = logging.getLogger(__name__)
 
+# A *deficit* exists when we have more jobs that can run on preemptable nodes than we have
+# preemptable nodes. In order to not block these jobs, we want to increase the number of non-
+# preemptable nodes that we have and need for just non-preemptable jobs. However, we may still
+# prefer waiting for preemptable instances to come available.
+#
+# To accomodate this, we set the delta to the difference between the number of provisioned
+# preemptable nodes and the number of nodes that were requested. when the non-preemptable thread
+# wants to provision nodes, it will multiply this delta times a preference for preemptable vs.
+# non-preemptable nodes.
+
```

Comment:
There was a blank line missing after the initialization of `_preemptableNodeDeficit` and I fixed that.